### ISSUE TITLE
[DSIP-33] Reduce WorkflowExecuteRunnable complexity

### ISSUE THREAD

**Issue by qingwli** [apache/dolphinscheduler#15901]
### Search before asking

- [X] I had searched in the [DSIP](https://github.com/apache/dolphinscheduler/issues/14102) and found no similar DSIP.

### Motivation

Improve WorkflowExecuteContext, reduce the number of database queries, and reduce the number of NPEs

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by ruanwenjun** [comment]
Good idea, this should be a huge task, and can be spitted into multiple task.
